### PR TITLE
feat: Autofill for credentials

### PR DESCRIPTION
**Pull request by borichellow**
# What's new in this PR?

- Autofill for credentials email/userName and password fields
- Also fixed not displaying `ServerVersionNotSupported` and `ClientUpdateRequired` dialogs in LoginScreen

### Issues

about the `ServerVersionNotSupported` and `ClientUpdateRequired` dialogs: they were not displayed when should be

### Causes (Optional)

In LoginScreen there are 3 Viewmodels: `LoginViewModel`, `LoginEmailViewModel` and `LoginSSOViewModel` and each of it has it's own `LoginState`. 
the state of `ServerVersionNotSupported` and `ClientUpdateRequired` was updated (call to show these dialogs) in `LoginEmailViewModel` and observed in `LoginScreen` which uses `LoginViewModel`. So updating in `LoginEmailViewModel` was always ignored.

### Solutions

1. Wrapped `LoginState` into `MutableStateFlow` instead of `MutableState` (just because I like it more and it can be reused in iOS later)
2. move `ServerVersionNotSupported` and `ClientUpdateRequired` states to `LoginError.DialogError` cause it makes more sense. And that error is already observed in `LoginEmailScreen`

### Attachments (Optional)

https://user-images.githubusercontent.com/6539347/192738580-aa800f08-2515-40ad-b632-8e87842f7ed4.mp4

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1047/3/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1047/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4540235)(4540235)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4540235)(4540235)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/common/textfield/AutoFillTextField.kt [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.unit.Dp
+import com.wire.android.ui.theme.wireDimensions
+import com.wire.android.ui.theme.wireTypography
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+internal fun AutoFillTextField(
```
Comment (MohamadJaara) [inline app/src/main/kotlin/com/wire/android/ui/common/textfield/AutoFillTextField.kt:30]: The issue with using the same Composable for login and register screens is now in register I get autofill from accounts I already have which is not the intended behaviour, for now, we can use it only for login and find a solution for register later

**Review by MohamadJaara — CHANGES_REQUESTED** [review]
nice work, only a comment about getting auto fill in register screen

File: app/src/main/kotlin/com/wire/android/ui/authentication/login/LoginViewModel.kt [hunk 2]
```diff
@@ -35,31 +34,37 @@ open class LoginViewModel @Inject constructor(
...
-        loginState = if (error is LoginError.None) {
-            loginState.copy(loginError = error)
+        val newState = if (error is LoginError.None) {
+            loginStateFlow.value.copy(loginError = error)
         } else {
-            loginState.copy(emailLoginLoading = false, loginError = error).updateEmailLoginEnabled()
+            loginStateFlow.value.copy(emailLoginLoading = false, loginError = error).updateEmailLoginEnabled()
         }
+        updateLoginState(newState)
     }
```
Comment (trOnk12) [inline app/src/main/kotlin/com/wire/android/ui/authentication/login/LoginViewModel.kt]: do not get me wrong, but I think the personal preferences to either use MUtableStateFlow of MutatbleStateOf, should not be taken as our collective decision, I have personally no strong on opinion which we are going to use, but this introduces chaos to a code, some places use mutableStateFlow some use MutableState, which one should we use ?I think if MutableStateFlow has some benefits over mutableStateOf we should take a decision on this during the collective and refactor piece by piece places we use the other option. Also ViewModel is not giong to be reused on iOS

Comment (borichellow) [reply app/src/main/kotlin/com/wire/android/ui/authentication/login/LoginViewModel.kt]: true story, it's better to discuss in Collective! 
I reverted `mutableStateFlow` changes for now :)

File: app/src/main/kotlin/com/wire/android/ui/common/textfield/AutoFillTextField.kt (hunk 1 above)
Comment (borichellow) [reply app/src/main/kotlin/com/wire/android/ui/common/textfield/AutoFillTextField.kt:30]: make sense, i removed password suggestion in createAcc screen

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1047/7/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4618697)(4618697)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4618697)(4618697)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-1047/8/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4643079)(4643079)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4643079)(4643079)-dev-release.apk)

**Review by MohamadJaara — APPROVED** [review]
LGTM 🚀

**Comment by AndroidBob** [comment]
Build [9](https://10.10.124.134/job/wire-android-reloaded/job/PR-1047/9/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4650965)(4650965)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4650965)(4650965)-dev-release.apk)

**Pull request merged**